dicom_validator/validator/iod_validator.py

```diff
@@ -199,13 +218,21 @@ def _object_is_required(self, condition):
 
     def _has_module(self, module_info):
         for tag_id_string, attribute in module_info.items():
-            if attribute['type'] != '3':
```

Comment:
Why has this been removed? I think we need type 1 or 2 tags for the check.

Reply:
If in the standard, a module has a few type 3 attributes and also some type 1 or 2 attributes, but the DCM file only has the type 3 attributes present then the validator should catch that and report error ("Some mandatory attributes are missing from this module"). Or so I thought :)

Reply:
Actually these would be reported anyway. The difference is only the handling if the module is not allowed - if there are only type 3 tags, it currently does not warn. I'm a bit undecided here. With the check, optional tags from modules that are not allowed would not be reported, without the check, they may be falsely reported if they also belong to other modules (which they tend to). Probably this needs some additional work outside of this PR.

Reply:
I tried to add some heuristic to detect if a module should be validated or not.
Basically, we validate a module only if it has attributes in the dataset that don't belong to any other module.

Reply:
Yes, that makes sense, I think. I will have a closer look later.

Reply:
Yes, this looks good - better than before, certainly.